Modules/gcmodule.c

```diff
@@ -462,6 +462,43 @@ visit_decref(PyObject *op, void *parent)
     return 0;
 }
 
+int tupletraverse(PyTupleObject *o, visitproc visit, void *arg);
+int func_traverse(PyFunctionObject *f, visitproc visit, void *arg);
+int cell_traverse(PyCellObject *op, visitproc visit, void *arg);
+int dict_traverse(PyObject *op, visitproc visit, void *arg);
+int list_traverse(PyListObject *o, visitproc visit, void *arg);
+
+/* Optimization:
+ * Calling inlined_tp_traverse(op, proc, c) is equivalent to calling
+ * Py_TYPE(op)->tp_traverse(op, proc, c), but is intended to be faster.  The
+ * approach is to check for common Python types and statically call their
+ * traverse functions (changing an indirect call to a static call), and by
+ * using a combination of the always_inline and flatten attributes to get the
+ * compiler to generate specializations of the traversal methods, getting rid
+ * of the indirect calls via Py_VISIT.
+ */
+__attribute__((always_inline)) __attribute__((flatten))
```

Comment:
Are these flags GCC or clang specific? We support many compilers so this likely need a bunch of macros guarding the compiler.

Reply:
Good point -- I believe they are supported by modern compilers but I'll add guards for older ones and for windows.